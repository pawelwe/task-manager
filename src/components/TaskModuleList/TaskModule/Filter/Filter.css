@charset "UTF-8";
.Filter {
  flex-basis: 50%;
  font-size: 1.1rem;
  padding: 0;
  margin: 1rem 0;
  font-weight: bold;
  color: crimson;
  text-align: left;
  background-color: transparent;
  appearance: none; }
  .Filter:last-of-type {
    margin-right: 0; }
  .Filter:hover {
    color: #f37c94; }

.isActive:after {
  content: 'Â«';
  margin-left: 5px; }

@media (min-width: 768px) {
  .Filter {
    flex-basis: 40%;
    margin: 0 1rem 0.4rem 0; } }
