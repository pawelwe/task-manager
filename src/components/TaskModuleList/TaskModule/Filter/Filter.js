import React from 'react';
import classes from './Filter.css';

const Filter = ({ sortBy, moduleId, filter, text, sortTasksBy }) => {
  return (
    <button
      id="filter"
      onClick={() => {
        sortBy(moduleId, filter);
      }}
      className={`${classes.Filter} ${sortTasksBy === filter ? classes.isActive : ''}`}
    >
      {text}
    </button>
  );
};

export default Filter;
