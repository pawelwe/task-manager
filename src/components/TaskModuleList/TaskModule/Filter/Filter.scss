@import '../../../App/scss/variables';

.Filter {
  flex-basis: 50%;
  font-size: 1.1rem;
  padding: 0;
  margin: 1rem 0;
  font-weight: bold;
  color: $accent;
  text-align: left;
  background-color: transparent;
  appearance: none;

  &:last-of-type {
    margin-right: 0;
  }
  &:hover {
    color: lighten($accent, 25%);
  }
}

.isActive {
  &:after {
    content: 'Â«';
    margin-left: 5px;
  }
}

@media (min-width: $tablet) {
  .Filter {
    flex-basis: 40%;
    margin: 0 1rem 0.4rem 0;
  }
}