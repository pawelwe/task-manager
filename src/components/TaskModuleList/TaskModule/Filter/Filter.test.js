import React from 'react';
import { mount } from 'enzyme';
import Filter from './Filter';

describe('Filter Component', () => {
  const sortBy = jest.fn();
  const wrapper = mount(<Filter sortBy={sortBy} />);

  it('Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('sortBy function', () => {
    wrapper.find('#filter').simulate('click');
    expect(sortBy).toHaveBeenCalled();
  });

});
