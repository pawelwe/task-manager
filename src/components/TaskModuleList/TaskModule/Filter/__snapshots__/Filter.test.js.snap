// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component Snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Filter
    sortBy={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "sortBy": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": <button
        id="filter"
        class="undefined undefined"
/>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "undefined undefined",
        "id": "filter",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        undefined,
      ],
      "type": "button",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "sortBy": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": <button
          id="filter"
          class="undefined undefined"
/>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "undefined undefined",
          "id": "filter",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          undefined,
        ],
        "type": "button",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
